add `soft_delete_retention_days` support for `azurerm_key_vault` (#7822)
terraform-providers/terraform-provider-azurerm: website/docs/r/key_vault.html.markdown

~> **Note:** Terraform will check when creating a Key Vault for a previous soft-deleted Key Vault and recover it if one exists. You can configure this behaviour using the `features` block within the `provider` block.  

> * `soft_delete_retention_days` - (Optional) The key Vault retention days when enable soft delete.

Reviewer: ```suggestion
* `soft_delete_retention_days` - (Optional) The number of days that items should be retained for once soft-deleted.
```

> !> **Note:** Once Soft Delete has been Enabled it's not possible to update it.

Reviewer: can we make this:

```suggestion
~> **Note:** This field can only be set once Soft Delete is Enabled and cannot be updated.
```
